core/templates/services/auth.service.ts

```diff
@@ -111,50 +112,74 @@ export class AuthService {
     }
   }
 
-  static getConfig: () => FirebaseOptions = function () {
-    let config: string = '';
-    let request = new XMLHttpRequest();
+  static async fetchConfigFromBackend(): Promise<FirebaseOptions> {
+    let config = '';
+    const defaultFirebaseConfig = {
+      FIREBASE_CONFIG_API_KEY: AppConstants.FIREBASE_CONFIG_API_KEY,
+      FIREBASE_CONFIG_AUTH_DOMAIN: AppConstants.FIREBASE_CONFIG_AUTH_DOMAIN,
+      FIREBASE_CONFIG_PROJECT_ID: AppConstants.FIREBASE_CONFIG_PROJECT_ID,
+      FIREBASE_CONFIG_STORAGE_BUCKET:
+        AppConstants.FIREBASE_CONFIG_STORAGE_BUCKET,
+      FIREBASE_CONFIG_MESSAGING_SENDER_ID:
+        AppConstants.FIREBASE_CONFIG_MESSAGING_SENDER_ID,
+      FIREBASE_CONFIG_APP_ID: AppConstants.FIREBASE_CONFIG_APP_ID,
+    };
     try {
-      request.open('GET', '/firebase_config', false);
-      request.send(null);
-      if (request.status === 200) {
-        let jsonResponse = request.responseText.substr(
-          request.responseText.indexOf(")]}'") + 4
-        );
+      const response = await fetch('/firebase_config');
+      if (response.ok) {
+        let jsonResponse = await response.text();
+        if (jsonResponse.startsWith(")]}'")) {
```

Comment:
Why do we need the "if" -- either it starts with the XSSI_PREFIX or it doesn't? Maybe worth explaining/simplifying.

Reply:
Added a comment to explain.